onnx/test/version_converter_test.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from onnx import checker, helper, ModelProto, TensorProto, GraphProto, NodeProto, OperatorSetIdProto
+from typing import Sequence, Text, Tuple, List, Callable
+from onnx import numpy_helper
+
+import numpy as np  # type: ignore
+
+import onnx.version_converter
+import unittest
+
+
+class TestVersionConverter(unittest.TestCase):
+
+    def _converted(
+            self,
+            graph,  # type: GraphProto
+            initial_version,  # type: OperatorSetIdProto
+            target_version  # type: int
+    ):  # type: (...) -> ModelProto
+        orig_model = helper.make_model(graph, producer_name='onnx-test', opset_imports=[initial_version])
+        # print(type(orig_model))
+        converted_model = onnx.version_converter.convert_version(orig_model,
+                target_version)
+        checker.check_model(converted_model)
+        return converted_model
+
+    # fn is a function that takes a single node as argument
+    def _visit_all_nodes_recursive(
+            self,
+            graph,  # type: GraphProto
+            fn  # type: Callable[[NodeProto], None]
+    ):  # type: (...) -> None
+        for node in graph.node:
+            fn(node)
+            for attr in node.attribute:
+                if attr.g is not None:
+                    self._visit_all_nodes_recursive(attr.g, fn)
+                if len(attr.graphs):
+                    for gr in attr.graphs:
+                        self._visit_all_nodes_recursive(gr, fn)
+
+    # Test 1: Backwards Incompatible Conversion: Add: 8 -> 2
```

Comment:
You probably want to make this case more complicated. Add has broadcast and axis attribute.

Reply:
This is all covered in the actual adapter tests on the adapters branch: https://github.com/onnx/onnx/pull/1184/files#diff-fe3eada8cd6b3b792277582934f39198

Reply:
My point is that let's have a case, in which the adapter is not visible.